test/common_utils.py

```diff
@@ -270,84 +270,57 @@ def combinations_grid(**kwargs):
+            # checking that `torch.testing.assert_close` does while giving us the ability to ignore the regular value
+            # comparison.
+            nonlocal value_comparison_failure
+            value_comparison_failure = True
+            if msg is None:
+                return default_msg
+            elif isinstance(msg, str):
+                return msg
+            elif callable(msg):
+                return msg(default_msg)
+            else:
+                raise pytest.UsageError(f"`msg` can be either be `None`, a `str` or a callable, but got {msg}.")
 
-        if self.mae:
-            if actual.dtype is torch.uint8:
-                actual, expected = actual.to(torch.int), expected.to(torch.int)
-            mae = float(torch.abs(actual - expected).float().mean())
-            if mae > self.atol:
-                self._fail(
-                    AssertionError,
-                    f"The MAE of the images is {mae}, but only {self.atol} is allowed.",
-                )
-        else:
-            super()._compare_values(actual, expected)
+        try:
+            return torch.testing.assert_close(
+                actual, expected, atol=atol, msg=msg_callback if enable_mae_comparison else msg, **kwargs
+            )
+        except AssertionError:
+            if not (value_comparison_failure and mae):
+                raise
 
+        if actual.dtype is torch.uint8:
+            actual, expected = actual.to(torch.int), expected.to(torch.int)
+        mae_value = float(torch.abs(actual - expected).float().mean())
+        if mae_value > atol:
+            raise AssertionError(f"The MAE of the images is {mae_value}, but only {atol} is allowed.")
 
-def assert_close(
-    actual,
-    expected,
-    *,
-    allow_subclasses=True,
-    rtol=None,
-    atol=None,
-    equal_nan=False,
-    check_device=True,
-    check_dtype=True,
-    check_layout=True,
-    check_stride=False,
-    msg=None,
-    **kwargs,
-):
-    """Superset of :func:`torch.testing.assert_close` with support for PIL vs. tensor image comparison"""
-    __tracebackhide__ = True
-
-    error_metas = not_close_error_metas(
-        actual,
-        expected,
-        pair_types=(
-            NonePair,
-            BooleanPair,
-            NumberPair,
-            ImagePair,
-            TensorLikePair,
-        ),
-        allow_subclasses=allow_subclasses,
-        rtol=rtol,
-        atol=atol,
-        equal_nan=equal_nan,
-        check_device=check_device,
-        check_dtype=check_dtype,
-        check_layout=check_layout,
-        check_stride=check_stride,
-        **kwargs,
-    )
+    actual_flat, actual_spec = tree_flatten(actual)
+    expected_flat, expected_spec = tree_flatten(expected)
+    assert actual_spec, expected_spec
 
-    if error_metas:
-        raise error_metas[0].to_error(msg)
+    for actual_item, expected_item in zip(actual_flat, expected_flat):
```

Comment:
The only "regression" compared to the patched `assert_close` is that we don't get any information where inside a sample an error happened. Consider the input sample to a detection model, i.e. a two-tuple with an image and a target dictionary. If the comparison of the bounding boxes target fails, `assert_close` would include `[1]["boxes"]` in the error message, whereas now we will only see the regular message and thus have to figure out ourselves what part of the input caused this.

We can re-implement this behavior if needed, but not sure its worth it. I guesstimate that 99% of our comparisons are single elements and thus no extra traceback is needed.

Apart from this, `assert_close_with_image_support` should be a faithful reproduction of the patched `assert_close` we had before.